### PR TITLE
Privacy component feature cleanup

### PR DESCRIPTION
**Pull request by allisonking**
Closes N/A

### Code Changes

Addresses the comments I left in https://github.com/ethyca/fides/pull/3411
* [x] Updates changelog for a few changes made that weren't recorded
* [x] Updates eslint rules
* [x] Moves the `hasMounted` check from `ConsentBanner` to `Overlay`
* [x] Refactor the accept all, reject all, and save preferences to be more dry
* [x] Prepend all CSS class names with `fides-`
* [x] Use `fides-js`'s `transformUserPreferenceToBoolean` and removes the duplicate one in privacy center

### Steps to Confirm

* Everything still works!

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### Description Of Changes

_Write some things here about the changes and any potential caveats_

### CODE REVIEW

File: clients/fides-js/src/components/ConsentModal.tsx [hunk 1]
```diff
@@ -59,24 +58,39 @@ const ConsentModal = ({
 
   return (
     // @ts-ignore A11yDialog ref obj type isn't quite the same
-    <div data-testid="consent-modal" {...container} className="modal-container">
-      <div {...overlay} className="modal-overlay" />
-      <div data-testid="modal-content" {...dialog} className="modal-content">
+    <div
```
Comment (allisonking) [inline clients/fides-js/src/components/ConsentModal.tsx:61]: this is all adding `fides-` to the beginning of every class name, but prettier adds new lines so the diff is more intense than it actually is!

File: CHANGELOG.md [hunk 2]
```diff
@@ -31,6 +31,11 @@ The types of changes are:
 - Load default privacy notices on startup [#3401](https://github.com/ethyca/fides/pull/3401/files)
 - Add privacy centre button text customisations [#3432](https://github.com/ethyca/fides/pull/3432)
 - Add privacy centre favicon customisation [#3432](https://github.com/ethyca/fides/pull/3432)
+- Add ability for `fides-js` to make API calls to Fides [#3361](https://github.com/ethyca/fides/pull/3361)
```
Comment (eastandwestwind) [inline CHANGELOG.md:34]: do we want to consolidate all the changelog items as part of this feat branch once we merge into `main`?

File: clients/privacy-center/components/consent/NoticeDrivenConsent.tsx [hunk 3]
```diff
@@ -6,6 +6,7 @@ import {
   getConsentContext,
   getOrMakeFidesCookie,
   saveFidesCookie,
+  transformUserPreferenceToBoolean,
```
Comment (eastandwestwind) [inline clients/privacy-center/components/consent/NoticeDrivenConsent.tsx:9]: nice!

File: clients/fides-js/src/components/Overlay.tsx [hunk 4]
```diff
@@ -86,88 +88,80 @@ const Overlay: FunctionComponent<OverlayProps> = ({
     [experience]
   );
 
+  const privacyNotices = useMemo(
+    () => experience.privacy_notices ?? [],
+    [experience.privacy_notices]
+  );
+
+  const handleUpdatePreferences = useCallback(
```
Comment (eastandwestwind) [inline clients/fides-js/src/components/Overlay.tsx:96]: thanks for consolidating this!

**Review by eastandwestwind — APPROVED** [review]
Overall looks good, haven't tested manually, but just a couple Qs

File: CHANGELOG.md (hunk 2 above)
Comment (eastandwestwind) [reply CHANGELOG.md:34]: I was under the impression we only document the PRs merged directly into `main` and not PRs as part of a feat branch, but either way works for me

Comment (allisonking) [reply CHANGELOG.md:34]: oh oops I didn't see your second comment until just now after I've merged 🤦‍♀️ let's ask the team!

**Pull request merged**